File: packages/firestore/ios/Plugin/Classes/Options/SetDocumentOptions.swift
import Foundation
import Capacitor
import FirebaseFirestore

@objc public class SetDocumentOptions: NSObject {
    private var reference: String
    private var data: [String: Any]
    private var merge: Bool

    init(reference: String, data: JSObject, merge: Bool, firestore: Firestore) {
        self.reference = reference
        self.data = FirebaseFirestoreHelper.createHashMapFromJSObject(data, firestore: firestore)
        self.merge = merge
    }

    func getReference() -> String {
        return reference
    }

    func getData() -> [String: Any] {
        return data
    }

    func getMerge() -> Bool {
        return merge
    }
}

File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/SetDocumentOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
import java.util.Map;
import org.json.JSONException;

public class SetDocumentOptions {

    private String reference;
    private Map<String, Object> data;
    private boolean merge;

    public SetDocumentOptions(
        String reference,
        JSObject data,
        boolean merge,
        @NonNull com.google.firebase.firestore.FirebaseFirestore firestore
    ) throws JSONException {
        this.reference = reference;
        this.data = FirebaseFirestoreHelper.createHashMapFromJSONObject(data, firestore);
        this.merge = merge;
    }

    public String getReference() {
        return reference;
    }

    public Map<String, Object> getData() {
        return data;
    }

    public boolean getMerge() {
        return merge;
    }
}
